mitiq/shadows/shadows.py

```diff
@@ -0,0 +1,198 @@
+# Copyright (C) Unitary Fund
+#
+# This source code is licensed under the GPL license (v3) found in the
+# LICENSE file in the root directory of this source tree.
+"""Classical shadow estimation for quantum circuits. Based on the paper"""
+
+from typing import (
+    Optional,
+    Callable,
+    List,
+    Dict,
+    Any,
+    Tuple,
+    Union,
+    Mapping,
+)
+
+import cirq
+import numpy as np
+from numpy.typing import NDArray
+
+import mitiq
+from mitiq import MeasurementResult
+from mitiq.shadows.quantum_processing import random_pauli_measurement
+from mitiq.shadows.classical_postprocessing import (
+    shadow_state_reconstruction,
+    expectation_estimation_shadow,
+    get_pauli_fidelities,
+)
+
+
+def pauli_twirling_calibrate(
+    k_calibration: int = 1,
+    locality: Optional[int] = None,
+    zero_state_shadow_outcomes: Optional[Tuple[List[str], List[str]]] = None,
+    qubits: Optional[List[cirq.Qid]] = None,
+    executor: Optional[Callable[[cirq.Circuit], MeasurementResult]] = None,
+    num_total_measurements_calibration: Optional[int] = 20000,
+) -> Dict[str, complex]:
+    r"""
+    This function returns the dictionary of the median of means estimation
+    of Pauli fidelities: {:math:`\{'b':f_{b}\}_{b\in\{0,1\}^n}`}.
```

Comment:
Can you clarify in the docstring if this dictionary will contain an entry for every $b\in\{0, 1\}^n$, or just a subset.

Reply:
The number of $f_b$ is related to the number of irreps in the PTM rep of the twirling group, and related to the locality of the interested operator, I added a reference for clarification.